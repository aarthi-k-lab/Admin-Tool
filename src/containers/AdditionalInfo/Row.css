.root>* {
    border-bottom: unset;
}

.rowStyle {
    font-size: 9.5px;
}

.statusDateStyle {
    font-size: 8px;
}

.setBackground {
    background: rgb(229, 248, 251);
}

.historyTable {
    padding-bottom: 0;
    padding-top: 0;
}