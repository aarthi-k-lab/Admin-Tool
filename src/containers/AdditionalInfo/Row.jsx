
import React from 'react';
import PropTypes from 'prop-types';
import Box from '@material-ui/core/Box';
import Collapse from '@material-ui/core/Collapse';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import * as R from 'ramda';
import './Row.css';

const formatDate = date => (date ? R.replace('T', ' ', date) : '-');

const Row = (props) => {
  const { data, onClick, changeColor } = props;
  const [open, setOpen] = React.useState(false);
  return (
    <React.Fragment>
      <TableRow className="root" onClick={onClick} styleName={changeColor ? 'setBackground' : ''}>
        <TableCell align="center" colSpan={1} component="th" scope="row">
          {data.evalId === 0 ? '-' : data.evalId}
        </TableCell>
        <TableCell align="center">{data.resolutionId}</TableCell>
        <TableCell align="center">{R.propOr('-', 'status', data)}</TableCell>
        <TableCell align="center" colSpan={2}>{data.statusDate}</TableCell>
        <TableCell align="center" colSpan={1}>{R.propOr('-', 'substatus', data)}</TableCell>
        <TableCell align="center" colSpan={2}>{data.substatusDate}</TableCell>
        {!R.isNil(data.evalHistory) && !R.isEmpty(data.evalHistory)
          ? (
            <TableCell
              align="center"
              onClick={() => setOpen(!open)}
              style={{ color: 'lightseagreen', whiteSpace: 'nowrap' }}
            >
              {open ? 'Hide History' : 'Show History '}
            </TableCell>
          )
          : <TableCell />}
      </TableRow>
      {open && (
      <TableRow>
        <TableCell colSpan={9} styleName="historyTable">
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box margin={1}>
              <Table aria-label="purchases" size="small">
                <TableHead style={{ background: '#8080802b' }}>
                  <TableRow>
                    <HistoryHeader label="CHANGE TYPE" />
                    <HistoryHeader label="APPROVAL TYPE" />
                    <HistoryHeader label="UPDATED DATE" />
                    <HistoryHeader label="USER NAME" />
                  </TableRow>
                </TableHead>
                <TableBody>
                  {data.evalHistory.map(historyRow => (
                    <TableRow key={historyRow.EvalId}>
                      <HistoryRow content={historyRow.ChangeType} />
                      <HistoryRow content={historyRow.ApprovalType} />
                      <HistoryRow content={formatDate(historyRow.UpdateDate)} />
                      <HistoryRow content={historyRow.UserName} />
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
      )}
    </React.Fragment>
  );
};

const HistoryHeader = (props) => {
  const { label } = props;
  return (<TableCell align="center" style={{ color: 'grey' }}>{label}</TableCell>);
};

HistoryHeader.propTypes = {
  label: PropTypes.string.isRequired,
};

const HistoryRow = (props) => {
  const { content } = props;
  return (<TableCell align="center" styleName="rowStyle">{content}</TableCell>);
};

HistoryRow.propTypes = {
  content: PropTypes.string.isRequired,
};


Row.propTypes = {
  changeColor: PropTypes.bool.isRequired,
  data: PropTypes.shape({
    evalHistory: PropTypes.arrayOf(
      PropTypes.shape({
        ApprovalType: PropTypes.string.isRequired,
        ChangeType: PropTypes.string.isRequired,
        UpdateDate: PropTypes.string.isRequired,
        UserName: PropTypes.string.isRequired,
      }),
    ).isRequired,
    evalId: PropTypes.string.isRequired,
    resolutionId: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    statusDate: PropTypes.string.isRequired,
    substatus: PropTypes.string.isRequired,
    substatusDate: PropTypes.string.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Row;
