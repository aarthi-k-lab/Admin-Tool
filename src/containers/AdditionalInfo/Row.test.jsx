import React from 'react';
import renderer from 'react-test-renderer';
import Row from './Row';

const onClick = jest.fn();
describe('renders <Row /> when changeColor is true', () => {
  const props = {
    onClick,
    data: {
      evalId: 'evalId',
      resolutionId: 'resolutionId',
      statusDate: '2015-11-10 03:19:42.4',
      status: 'status',
      substatus: 'subStatus',
      loanNumber: 'loanNumber',
      substatusDate: '2015-11-10 03:19:42.4',
      evalHistory: [{
        EvalId: 'EvalId',
        ChangeType: 'ChangeType',
        ApprovalType: 'ApprovalType',
        UpdateDate: 'UpdateDate',
        UserName: 'UserName',
      }],
    },
    changeColor: true,
  };
  test('snapshot test', () => {
    const tree = renderer
      .create(<Row {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('renders <Row /> when changeColor is false', () => {
  const props = {
    onClick,
    data: {
      evalId: 'evalId',
      resolutionId: 'resolutionId',
      statusDate: '2015-11-10 03:19:42.4',
      status: 'status',
      substatus: 'subStatus',
      loanNumber: 'loanNumber',
      substatusDate: '2015-11-10 03:19:42.4',
      evalHistory: [{
        EvalId: 'EvalId',
        ChangeType: 'ChangeType',
        ApprovalType: 'ApprovalType',
        UpdateDate: 'UpdateDate',
        UserName: 'UserName',
      }],
    },
    changeColor: false,
  };
  test('snapshot test', () => {
    const tree = renderer
      .create(<Row {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
