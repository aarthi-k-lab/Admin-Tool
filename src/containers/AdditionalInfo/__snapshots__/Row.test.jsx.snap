// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Row /> when changeColor is false snapshot test 1`] = `
<tr
  className="MuiTableRow-root root "
  onClick={[MockFunction]}
  role={null}
>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={1}
    role="cell"
    scope="row"
  >
    evalId
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    resolutionId
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    status
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={2}
  >
    Invalid date
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={1}
  >
    subStatus
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={2}
  >
    Invalid date
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    onClick={[Function]}
    style={
      Object {
        "color": "lightseagreen",
        "cursor": "pointer",
        "whiteSpace": "nowrap",
      }
    }
  >
    Show History 
  </td>
</tr>
`;

exports[`renders <Row /> when changeColor is true snapshot test 1`] = `
<tr
  className="MuiTableRow-root root containers-AdditionalInfo-Row__setBackground"
  onClick={[MockFunction]}
  role={null}
>
  <th
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={1}
    role="cell"
    scope="row"
  >
    evalId
  </th>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    resolutionId
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
  >
    status
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={2}
  >
    Invalid date
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={1}
  >
    subStatus
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    colSpan={2}
  >
    Invalid date
  </td>
  <td
    aria-sort={null}
    className="MuiTableCell-root MuiTableCell-alignCenter"
    onClick={[Function]}
    style={
      Object {
        "color": "lightseagreen",
        "cursor": "pointer",
        "whiteSpace": "nowrap",
      }
    }
  >
    Show History 
  </td>
</tr>
`;
